phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixStatement.java

```diff
@@ -59,4 +70,17 @@ public ResultSet getGeneratedKeys() throws SQLException {
         return new LoggingPhoenixResultSet(super.getGeneratedKeys(), phoenixMetricsLog, this.sql);
     }
 
+    private void loggingAutoCommitHelper() throws SQLException {
+        Connection conn = this.getConnection();
+
+        if(stmt.getUpdateOperation().isMutation() && conn.getAutoCommit()){
+            phoenixMetricsLog.logWriteMetricsfoForMutationsSinceLastReset(
```

Comment:
Can you reuse this piece of code from somewhere?

Reply:
Unlikely, but let me check that.

Reply:
Looks like we can't.

Reply:
Can you check `LoggingPhoenixConnection`? It has similar calls multiple times as well. Might be good to refactor this along with it.